### ISSUE TITLE
2.1.3 alpha issues

### ISSUE THREAD

**Issue by RyanTG** [scottwainstock/pbm-android#54]
**Bugs:**
App crashes when removing machine from location. Details: when clicking the trash icon in the menu, nothing happens; when clicking the "remove" text at the bottom, the app crashes. I replicated it twice.

**Issues:**
Events are not displaying the location name. Should show location name, if that field has been filled in. Event Title @ Location Name.

Display region name in more places:

* Top menu. Currently just says "Pinball Map" would be nice to say "Pittsburgh Pinball Map" just to remind users which region they're in.

* Contact admin. Change "send a message to your region's admin" to "send a message to Pittsburgh admin" (or whatever current region is).

* Suggest a new location. Show the region name.

Just like the recent additions shows the city name, I wonder if it would be appropriate to show the city name in the results of lookup by location type, machine, and zone. I think it may make those results more helpful.

**Comment by bdols** [comment]
do you have an example region where an event with a location already exists?

**Comment by RyanTG** [comment]
Ya, Portland region. The "POP Superleague" tournament is tagged with a location.

**Comment by bdols** [comment]
I mean, in staging.

**Comment by bdols** [comment]
I've fixed the bug and added the region name to where you specified. Not yet checked in.

The city name - I'd have to investigate some.

**Comment by RyanTG** [comment]
I don't know in staging, and can't look at the moment. I'll check when I get home. You could log in to Austin (unless you're not a user!) and add an event.

Thanks, Brian!

**Comment by RyanTG** [comment]
I'm home. On staging, the Los Angeles region has an event with a location. The location is called Casa De Carlos.

**Comment by bdols** [comment]
ok how much info on the location should be in the events listing?
Just the name?

**Comment by bdols** [comment]
![image](https://cloud.githubusercontent.com/assets/286836/8716541/f62da61c-2b57-11e5-924e-1fa0e6ddd91b.png)

**Comment by bdols** [comment]
Cities inlined in the LocationLookupDetail views:

![image](https://cloud.githubusercontent.com/assets/286836/8716757/eca031bc-2b59-11e5-9acc-71cefa5d7571.png)

**Comment by bdols** [comment]
more: 

![image](https://cloud.githubusercontent.com/assets/286836/8716796/20d27544-2b5a-11e5-970a-396b94de9fda.png)

![image](https://cloud.githubusercontent.com/assets/286836/8716807/3992de20-2b5a-11e5-8d9b-cbef64c5814b.png)

![image](https://cloud.githubusercontent.com/assets/286836/8716818/45c1e6c8-2b5a-11e5-9693-2bd5880f699e.png)

![image](https://cloud.githubusercontent.com/assets/286836/8716824/551ac860-2b5a-11e5-8d73-00d70cf9dd19.png)

**Comment by RyanTG** [comment]
Looks great! How's it look on a smaller display, though? Does the city name squish into the miles?

**Comment by bdols** [comment]
as of now, the distance is 100dip and cities are 200dip, this test 4.1.1 device has 480x800 pixels:

![image](https://cloud.githubusercontent.com/assets/286836/8741643/b20253f0-2c21-11e5-9303-a11d35fd55fe.png)

**Comment by bdols** [comment]
what it looks like (on a larger device) when location services are not available:

![image](https://cloud.githubusercontent.com/assets/286836/8741652/cd717198-2c21-11e5-83af-a65aa8265487.png)

**Comment by bdols** [comment]
small changes to the layout: made the distance field wider to accommodate users who are browsing further away and made the location name spaced away from the numMachines indicator:

![image](https://cloud.githubusercontent.com/assets/286836/8752474/de6799a2-2c79-11e5-9106-6a8d56d6d7aa.png)

**Comment by RyanTG** [comment]
When editing location info, the beta allows users to edit the "website" field. I think we should remove this ability, and make it read-only. For the iOS app, we determined that this could be a source of spam. So only admins can edit websites.

**Comment by RyanTG** [comment]
Is there a filter on Events? The Portland region is showing only one event (but there many: http://pinballmap.com/portland/events). The Los Angeles region is showing no events, but there should be two.

On my phone, on the main screen I see just one icon at the top: the question mark. The rest of the settings are under the triple-dot. I think it looks strange to have a single question mark icon. Can we put this link (contact region admin) within the triple-dot instead? Other views that only show the question mark: Recently Added; Events; Recent High Scores.

The Search icon (the microscope) seems small. I was just using the ebay app, and noticed that there's is a better size. Not sure if Android provides a stock icon.

**Comment by bdols** [comment]
the crashing that you describe shouldn't be happening, I put a lot of effort into making it crash much less. can you send me logcat output when it happens? If it is a crash, you should see a crash dialog and be able to submit a bug report and have the bug appear in the Android Developer console, which may suffice instead of logcat output.

the events are not showing because location_id was null for a bunch of events. I have a fix for this.

I've moved the "contact region admin" button down in priority, so it should appear lower in the menu.

that is the default android search icon, are you referring to the lookup screens?

**Comment by RyanTG** [comment]
I'll send you a log if I can replicate it. However, I just tried and could not replicate it! I'm switching between apps and it's smooth. I had just restarted my phone, though. I'll try again later in the day. 

Thanks!

**Comment by bdols** [comment]
there is code to reload all data for the zone after it expires. it supposed to be 1 day later after the first app start that it will reload, but I left it at 1 hour in the last build.. so I'll adjust that too..

**Comment by RyanTG** [comment]
And yeah, I was referring to the Search icon the lookup screens. But, let's forget I said anything. I just thought it looked a little small compared with the globe icon. But I'm used to it now. And you've done so much!

**Comment by RyanTG** [comment]
Note: I flashed Sailfish OS on my phone and can't test Android apps anymore. At some point ported installations of Sailfish will allow Android apps. But not yet!

**Comment by scottwainstock** [comment]
What's our fin checklist here for releasing this? I can help fix bugs if we
run into something weird.

**Comment by RyanTG** [comment]
Not sure. The app's looking good to me, though!

I know this isn't a bug, but I would love to see "lookup by city" added to this release. Scott, would you have any time to add that?

**Comment by bdols** [comment]
I've pushed fixes for the events bug and relocating the help/contact icon into my lollipop branch.
I've also upgraded to Android Studio 1.3.x and pushed some build changes for the IDE that may require that version of the IDE, namely around the underlying gradle version upgrade. Let me know if that's a problem for you, as it isn't a necessary change to get the bug fixes incorporated for this release.

**Comment by RyanTG** [comment]
I'll actually have Android on another device for a couple more days, if Scott is able to merge the updates.

**Comment by scottwainstock** [comment]
@bdols That's no problem. If you send a pull request for that, I can merge it, get it out, and we can consider it the final pre-release beta? Hopefully we can promote it/release it this weekend.

I can add lookup by city, yes. That's not going to happen in the immediate future though. Work is pretty bad right now. But I'll put it high on my list, and I'd rather not delay releasing Brian's stuff any longer.

**Comment by scottwainstock** [comment]
@bdols @RyanTG Brian's latest changes have been merged and promoted to the beta apk. Ryan, we can talk about it this weekend, but if possible I'd like to get this release out Sunday or Monday.

**Comment by RyanTG** [comment]
Sounds killer. I'm prepared to talk.

**Comment by bdols** [comment]
on my device, the name of the region in the toolbar at the top is obscured and abbreviated by the two buttons for Suggest Location, and Change Region. is this OK with you?

**Comment by RyanTG** [comment]
It's a little bothersome, because "suggest a location" fills up so much space. Is it possible to have all of those options within the triple dot menu, and not listed in the header? But keep the search and globe icons on the pages which currently show those in the header? 

If not, I think we should just return the contact region admin icon to where it was before. However, is it possible to use the "Help" icon from here: https://www.google.com/design/icons/ ? It's still a question mark, but it has a circle border, which I think is more obvious that it's an icon and not an error. Or are we not using this icon set?

**Comment by RyanTG** [comment]
Or the"live help" icon, maybe?

**Comment by bdols** [comment]
here's what "live help" looks like:

![pbm-toolbar1](https://cloud.githubusercontent.com/assets/286836/9400068/4fa4b8c0-4782-11e5-8148-325f6a6a1b74.png)

and with the "suggest" button still there. this is the nexus 7 emulator, so it may still abbreviate the app title:
![pbm-toolbar2](https://cloud.githubusercontent.com/assets/286836/9400078/7fee802e-4782-11e5-84cc-df542059dde2.png)

**Comment by RyanTG** [comment]
Nice! Thanks Brian. I like that bottom one. So on smaller screens the "suggest a location" would probably go under the three dots, so the region name wouldn't be too covered up?

**Comment by bdols** [comment]
it isn't, this is what it looks like on a smaller screen:

![image](https://cloud.githubusercontent.com/assets/286836/9400462/10244468-4787-11e5-9c01-3c398ee0c91d.png)

**Comment by bdols** [comment]
without the suggest button, looks better:

![image](https://cloud.githubusercontent.com/assets/286836/9400541/f98db83c-4787-11e5-8e39-9fc248712d5d.png)

**Comment by bdols** [comment]
submitted pull request with that change

**Comment by scottwainstock** [comment]
@bdols @RyanTG Hi Brian, Ryan found a repeatable bug that we want to fix before things go live. I'm happy to do the fix, but it might take me a little bit to get to it. If you get time before I do, the log is below.

"occurs only when you search for a location, then view that location's info, then press back. if you don't search, it doesn't crash when you press back. android 4.4, moto g"

java.lang.IndexOutOfBoundsException: Invalid index 5, size is 5
at java.util.ArrayList.throwIndexOutOfBoundsException(ArrayList.java:255)
at java.util.ArrayList.get(ArrayList.java:308)
at com.pbm.LocationListAdapter.getView(LocationListAdapter.java:48)
at android.widget.AbsListView.obtainView(AbsListView.java:2255)
at android.widget.ListView.measureHeightOfChildren(ListView.java:1263)
at android.widget.ListView.onMeasure(ListView.java:1175)
at android.view.View.measure(View.java:16540)
at android.widget.RelativeLayout.measureChild(RelativeLayout.java:689)
at android.widget.RelativeLayout.onMeasure(RelativeLayout.java:473)
at android.view.View.measure(View.java:16540)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5137)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
at android.support.v7.internal.widget.ContentFrameLayout.onMeasure(ContentFrameLayout.java:124)
at android.view.View.measure(View.java:16540)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5137)
at android.support.v7.internal.widget.ActionBarOverlayLayout.onMeasure(ActionBarOverlayLayout.java:444)
at android.view.View.measure(View.java:16540)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5137)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
at android.view.View.measure(View.java:16540)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5137)
at android.widget.LinearLayout.measureChildBeforeLayout(LinearLayout.java:1404)
at android.widget.LinearLayout.measureVertical(LinearLayout.java:695)
at android.widget.LinearLayout.onMeasure(LinearLayout.java:588)
at android.view.View.measure(View.java:16540)
at android.view.ViewGroup.measureChildWithMargins(ViewGroup.java:5137)
at android.widget.FrameLayout.onMeasure(FrameLayout.java:310)
at com.android.internal.policy.impl.PhoneWindow$DecorView.onMeasure(PhoneWindow.java:2291)
at android.view.View.measure(View.java:16540)
at android.view.ViewRootImpl.performMeasure(ViewRootImpl.java:1942)
at android.view.ViewRootImpl.measureHierarchy(ViewRootImpl.java:1132)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:1321)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1019)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:5725)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:761)
at android.view.Choreographer.doCallbacks(Choreographer.java:574)
at android.view.Choreographer.doFrame(Choreographer.java:544)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:747)
at android.os.Handler.handleCallback(Handler.java:733)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:136)
at android.app.ActivityThread.main(ActivityThread.java:5086)
at java.lang.reflect.Method.invokeNative(Native Method)
at java.lang.reflect.Method.invoke(Method.java:515)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:785)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:601)
at dalvik.system.NativeStart.main(Native Method)

**Comment by scottwainstock** [comment]
@bdols Tabs/spaces aside, what do you think of... https://github.com/scottwainstock/pbm-android/commit/23cbe956afae71b3ec0629d08fa19886dd280d32 ? This fixes the issue listed above. Any huge side effects here?

**Comment by bdols** [comment]
I don't recall exactly why the adapter.notifyDataSetChanged() call was in there, perhaps performance as the adapter should have the data already in memory?, I'd have to look in my notes.
But ClosestLocations does the same thing as your fix does, and the search filter works there when navigating back and forth.

BTW, I believe I'm using the default java formatting settings that Android Studio has.

**Comment by scottwainstock** [comment]
@bdols Gotcha. I'll update the formatting. I'll make a note of a potential performance issue here.

@bdols @RyanTG I'm going to send Ryan the most recent APK. I've got a thing Thursday night and then I'll be out of communication Friday-Saturday. So, I'm afraid to toss a release out before then. Any objections to SUNDAY night? Sorry to keep delaying this, I just want to be around and available the first couple days after it goes live.

**Comment by RyanTG** [comment]
Do either of you know why this is one of the app permissions? "modify or delete the contents of your USB storage. read the contents of your USB storage."

**Comment by bdols** [comment]
I asked about this on stackoverflow and got a good answer:
http://stackoverflow.com/questions/30809815/why-does-play-services-location-need-the-android-permission-write-external-stora
those permissions aren't needed and can be removed as stated

**Comment by RyanTG** [comment]
Closest locations should show more locations. Right now it looks like it's limited to 10 miles. Please expand it to 35 miles (10 miles isn't too helpful in giant regions like Los Angeles).

**Comment by bdols** [comment]
It's hard-coded in CloseLocations.java to 10 along with maxNumMachinesToDisplayOnMap is set to 25.
So, the number of machines displayed on the map should be increased as well.

I can make these changes if you want, along with the permissions change.

**Comment by RyanTG** [comment]
Please, thank you. Let's do 20 miles and 100 locations. Is that cool?

**Comment by scottwainstock** [comment]
Alright, holding off on the release until this goes in then...

**Comment by bdols** [comment]
![image](https://cloud.githubusercontent.com/assets/286836/9611944/66f51748-50a8-11e5-9261-db0ac5a9ee6f.png)

**Comment by bdols** [comment]
The manifest already contained WRITE_EXTERNAL_STORAGE. I tried removing it and the embedded map would crash because it needs external storage (on 5.1). The READ_EXTERNAL_STORAGE is new to play-services and I've removed that from the manifest. Haven't seen an issue on 4.1 and 5.1 test devices when remove READ_EXTERNAL.

Submitted pull request. Sorry about the reformatting of the AndroidManifest.xml file.
